core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -419,22 +437,17 @@ public static void udpateUserRoleMappingInBatchModeForInternalRoles(Connection d
                 log.debug(errorMessage, e);
             }
             throw new UserStoreException(errorMessage, e);
-        } finally {
-            if (localConnection) {
-                DatabaseUtil.closeAllConnections(dbConnection);
-            }
-            DatabaseUtil.closeAllConnections(null, prepStmt);
+        } finally {  // can remove since this is not needed with try-with-resources -
+            close(dbConnection);
```

Comment:
Since the connection getting passed as a parameter to this method, it shouldn't close the connection inside this method.

Need to close the connection only `localConnection` set to true.